criu/protobuf.c

```diff
@@ -25,7 +25,7 @@ static char *image_name(struct cr_img *img)
 	int fd = img->_x.fd;
 	static char image_path[PATH_MAX];
 
-	if (read_fd_link(fd, image_path, sizeof(image_path)) > 0)
+	if (fd >= 0 && read_fd_link(fd, image_path, sizeof(image_path)) > 0)
```

Comment:
Maybe we can do something like:

```
if (lazy_image(img)) {
        return img->path;
} else if (!empty_image(img)) {
        if (read_fd_link(fd, image_path, sizeof(image_path)) > 0)
                return image_path;
}

return NULL;
```

Reply:
Excellent suggestion. I made some modifications. I still want to do a `fd >= 0` test just in case we add new image failures.